pytorch_lightning/core/lightning.py

```diff
@@ -1615,11 +1621,31 @@ def save_hyperparameters(self, *args, frame=None) -> None:
         "p1": 1
         "p2": abc
         "p3": 3.14
+
+        >>> class ManuallyArgsModel(LightningModule):
+        ...     def __init__(self, arg1, arg2, arg3):
+        ...         super().__init__()
+        ...         # pass argument(s) to ignore as a string or in a list
+        ...         self.save_hyperparameters(ignore="arg2")
+        ...     def forward(self, *args, **kwargs):
+        ...         ...
+        >>> model = ManuallyArgsModel(1, 'abc', 3.14)
+        >>> model.hparams
+        "arg1": 1
+        "arg3": 3.14
         """
         if not frame:
             frame = inspect.currentframe().f_back
         init_args = get_init_args(frame)
         assert init_args, "failed to inspect the self init"
+
+        if ignore:
+            if isinstance(ignore, str):
+                ignore = [ignore]
+            if isinstance(ignore, list):
+                ignore = [arg for arg in ignore if isinstance(arg, str)]
+            init_args = {k: v for k, v in init_args.items() if k not in ignore}
```

Comment:
how about if `ignore=None` from default?
```
1 not in None <- TypeError: argument of type 'NoneType' is not iterable
```

Reply:
@Borda: I am doing a check for `if ignore`, if it's None, `init_args` won't be changed.

Reply:
And also the other tests will fail, as the default is None :)